py/server/deephaven/updateby.py

```diff
@@ -0,0 +1,276 @@
+    """Creates an EMA (exponential moving average) UpdateByClause for the supplied column names, using ticks as
+    the decay unit.
+
+    The formula used is
+        a = e^(-1 / time_scale_ticks)
+        ema_next = a * ema_last + (1 - a) * value
+
+    Args:
+        time_scale_ticks (int): the decay rate in ticks
+        cols (Union[str, List[str]]): the column(s) to be operated on, can be renaming expressions, i.e. "new_col = col"
+        ema_control (EmaControl): defines how special cases should behave, when None, the default EmaControl settings
+            as specified in :meth:`~EmaControl.__init__` will be used
+
+    Returns:
+        an UpdateByClause
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        if ema_control is None:
+            return UpdateByClause(j_updateby_clause=_JUpdateByClause.Ema(time_scale_ticks, *cols))
+        else:
+            return UpdateByClause(
+                j_updateby_clause=_JUpdateByClause.Ema(ema_control.j_ema_control, time_scale_ticks, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a tick-decay EMA UpdateByClause.") from e
+
+
+def ema_time_decay(ts_col: str, time_scale: Union[int, str], cols: Union[str, List[str]],
+                   ema_control: EmaControl = None) -> UpdateByClause:
+    """Creates an EMA(exponential moving average) UpdateByClause for the supplied column names, using time as the
+    decay unit.
+
+    The formula used is
+        a = e^(-dt / time_scale_nanos)
+        ema_next = a * ema_last + (1 - a) * value
+
+     Args:
+        ts_col (str): the column in the source table to use for timestamps
+
+        time_scale (Union[int, str]): the decay rate, can be expressed as an integer in nanoseconds or a time
+            interval string, e.g. "00:00:00.001"
+        cols (Union[str, List[str]]): the column(s) to be operated on, can be renaming expressions, i.e. "new_col = col"
+        ema_control (EmaControl): defines how special cases should behave,  when None, the default EmaControl settings
+            as specified in :meth:`~EmaControl.__init__` will be used
+
+    Returns:
+        an UpdateByClause
+
+    Raises:
+        DHError
+     """
+    try:
+        if isinstance(time_scale, str):
+            time_scale = _JDateTimeUtils.expressionToNanos(time_scale)
+        cols = to_sequence(cols)
+        if ema_control is None:
+            return UpdateByClause(j_updateby_clause=_JUpdateByClause.Ema(ts_col, time_scale, *cols))
+        else:
+            return UpdateByClause(
+                j_updateby_clause=_JUpdateByClause.Ema(ema_control.j_ema_control, ts_col, time_scale, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a time-decay EMA UpdateByClause.") from e
+
+
+def cum_sum(cols: Union[str, List[str]]) -> UpdateByClause:
+    """Creates a CumSum (cumulative sum) update-by clause for the supplied column names.
```

Comment:
Is there a reason that we want `CumSum` and such in these docs, instead of just having:
```suggestion
    """Creates a cumulative sum update-by clause for the supplied column names.
```
Is `CumSum` supposed to have some significant meaning here?  It feels like it is just a reference to a Java type.  If so, are we keeping this to make it easier to search the Javadocs?  Did we do the same in `agg`?